### ISSUE TITLE
Add "road condition"

### ISSUE THREAD

**Issue by hschoen** [OpenSimulationInterface/open-simulation-interface#60]
Did I miss it or don't we have a parameter for the road condition?

- wet
- icy
- covered with snow
- oil
- sand
- leaves
- etc.

Also the layer thickness would be relevant (especially water layer for radar).
Coverage (in percent) of road?

**Comment by TimoHankeBMW** [comment]
No such parameter exists yet as there is no exactly specified and agreed-upon list of road conditions.
Especially the 'exactly specified' part makes this quite hard, e.g. 'wet' is a very broad und unspecific term, of whichi every usage instance will likely have a different understanding.
As discussions for specifications would likely fill many days' worth of workshops, I'd suggest that the best way is to wait for a specification from the scenario definition and use that. If these don't match, we would need to adapt anyway.

**Comment by PhilippRoethigBMW** [comment]
@hschoen and @TimoHankeBMW what do you think about implementing the condition of the road in a separate message? This message could define specific and measurable properties of the road rather than just determine a general category for it. Of course we would need to decide which properties would make sense.

```cmd
//
// \brief The condition of the roads surface.
//
message RoadCondition
{
    // The temperature of the roads surface in degree Celsius.
    //
    optional double surface_temperature = 1;

    // The height of the water film on top of the surface in mm.
    //
    optional double surface_water_film = 2;

    // The temperature where the water on top of the surface would start to freeze or dew in degree Celsius.
    //
    optional double surface_freezing_point = 3;

    // The percentage of ice covering the road.
    //
    optional double surface_ice = 4;

    // The coefficient presenting the friction and the grip of the road.
    //
    optional double surface_friction = 5;
}
```

**Comment by hschoen** [comment]
@PhilippRoethigBMW, I think this is a good start! We should however merge this solution with the specification from the scenario definition as proposed by @TimoHankeBMW.
From a radar point of view I would like to add roughness of the road surface (clutter). Here we should use the definition used in environment simulation. @AndreasVIRES and @MHerrmannIPG , do you have a definition for roughness of the road surface?

**Comment by MHerrmannIPG** [comment]
@hschoen, no, there's currently no parameter for the roughness of the road surface. It's no problem to add a value to the test specification globally for the complete road network which then is available to any sensors. Making it variable along the track/in various areas of the network would be a bigger task.

**Comment by AndreasVIRES** [comment]
**roughness**
In OpenDRIVE there is a roughness value (without unit) in the Lane Material Record, I'll try to find out how this value typically is used (maybe mainly for sound simulation). Additionally OpenCRG can provide a very detailed "rough" surface.

**road conditions**
I would not recommend waiting for scenario definition. OpenSCENARIO is available in version 0.9 and we almost totally got rid of a road condition description and switched to a user specific effect/intensity list which for sure can contain something like "height of the water film on top of the surface" as an effect and the value in [mm] as intensity.
OpenDRIVE also allows setting an application specific surface material code and a friction value for lanes.

**Comment by hschoen** [comment]
@CarloVanDriestenBMW I think every sensor can estimate the road condition to a certain extent given it has a designated model for this purpose. Especially when the sensor points directly down to the surface of the road without any other objects in its FoV, an estimation of the road conditions should be possible. So I agree that we should define a designated message in OSI:Sensor(Output)Data.

**Comment by CarloVanDriestenBMW** [comment]
Could you make a proposal?

**Comment by carsten-kuebler** [comment]
@hschoen and @CarloVanDriestenBMW I add the new message OSI:RoadConditon as an attribute to the message OSI:Lane. So every message OSI:DetectedLane has its own road condition value. I think, we need no additional message in OSI:Sensor(Output)Data. Because OSI:Sensor(Output)Data holds OSI:DetectedLanes as an array.

**Comment by hschoen** [comment]
@pmai and @CarloVanDriestenBMW : We defined **roughness** as a parameter for road condition. However, we must distinguish between **fine roughness** (rf) and **unevenness** (ru). The total roughness (r) is the product of both: r=rf*ru.
The new parameter _surface_roughness_ actually corresponds to the unevenness index which is directly related to the unevenness. Hence we have to define an additional parameter _surface_fine_roughness_SD_ which is the **standard height deviation of fine roughness** [m].

**Comment by pmai** [comment]
@hschoen, @carsten-kuebler  does the definition in #133 fit the bill?

**Comment by carsten-kuebler** [comment]
@pmai I would agree, that additional values should be added in future PR.

**Comment by CarloVanDriestenBMW** [comment]
@carsten-kuebler I am not happy with that definition here:

> 178         // The coefficient presenting the friction and the grip of the road. 
> 179         // 
> 180         optional double surface_friction = 5;

**Comment by carsten-kuebler** [comment]
@CarloVanDriestenBMW I suggest to use the definition and reference values "Maximaler Kraftschlussbeiwert"
[Wikipedia link in German](https://de.wikipedia.org/wiki/Reibungskoeffizient)

**Comment by carsten-kuebler** [comment]
@PhilippRoethigBMW and @CarloVanDriestenBMW do we need surface friction for OSI sensor simulation at the moment (for Version 3.0.0)? There are a lot of other missing attributes to simulate 0D longitudinal dymamics of a vehicle.

**Comment by CarloVanDriestenBMW** [comment]
@carsten-kuebler we will remove it for now! I will merge your PR on that. Thanks

**Comment by carsten-kuebler** [comment]
@CarloVanDriestenBMW I made PR #152

**Comment by CarloVanDriestenBMW** [comment]
Bump: @hschoen do we meet your requirements now?

**Comment by carsten-kuebler** [comment]
@hschoen and @CarloVanDriestenBMW can we close this issue?

**Comment by hschoen** [comment]
@CarloVanDriestenBMW and @carsten-kuebler : I appreciate that with PR #152 we have now two parameters describing surface roughness (roughness and texture). However, I have problems with their definitions: 
First, surface_roughness is defined as the International Roughness Index (IRI) [mm/m] according to [1] which is not identical to the surface unevenness [cm³] as defined in [2]. I am not sure, if one can be converted into the other.
Second, surface_texture also describes fine roughness but is not clearly defined. I would suggest to use the standard height deviation s,h,(f) [mm] as given in [2] (5.33). For camera, a clear definition of surface texture is also missing.

**Comment by carsten-kuebler** [comment]
@pmai Could you check the surface_texture definition?

**Comment by CarloVanDriestenBMW** [comment]
bump @pmai

**Comment by CarloVanDriestenBMW** [comment]
As discussed the last Pegasus meeting, @hschoen will add a PR to adjust the documentation of the definition in order to be correct including the feedback from the other users.

**Comment by LudwigFriedmannBMW** [comment]
Hello @hschoen, will you provide a pull request for this issue for release 3.0.1? Or shall we postpone it to 3.1.0?

**Comment by hschoen** [comment]
@LudwigFriedmannBMW : As we do not have a common sense on the definition of radar related roughness I would suggest to postpone this to 3.1.0.

**Comment by LudwigFriedmannBMW** [comment]
@hschoen OSI 3.1.0 release date is coming closer.

How can we come up with a common solution?

**Comment by hschoen** [comment]
I do not agree that there is no further need for this issue. I do see that nobody is currently tackling this issue. As for me I will not be able to contribute this year. 
If it is a problem to keep this issue open we can reopen it when there are again activities on that issue.

**Comment by LudwigFriedmannBMW** [comment]
Well, theres no need to close the issue in that case. I'll postpone it to OSI 4.0.0
